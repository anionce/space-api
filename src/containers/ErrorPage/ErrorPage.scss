.error-container {
	display: flex;
	flex-direction: column;
	justify-items: center;
	align-items: center;
	gap: 0;

	img {
		width: 300px;
	}
}
