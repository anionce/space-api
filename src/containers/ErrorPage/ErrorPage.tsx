import './ErrorPage.scss';
import error from '../../assets/img/error.svg';
import React from 'react';

export const ErrorPage = () => {
	return (
		<div className='error-container'>
			<h1>Ooops!</h1>
			<h2>There has been an error.</h2>
			<h3>Please try again later...</h3>
			<img alt='error' src={error}></img>
		</div>
	);
};
