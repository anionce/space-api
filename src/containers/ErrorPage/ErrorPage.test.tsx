import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { screen } from '@testing-library/react';
import { customRender } from '../../assets/tests/utils';
import { ErrorPage } from './ErrorPage';

test('renders correcly', () => {
	const { container } = customRender(<ErrorPage />);

	expect(container).toMatchSnapshot();
});

test('all elements are displayed', () => {
	customRender(<ErrorPage />);

	expect(screen.getByText(/Ooops!/i)).toBeInTheDocument();
	expect(screen.getByText(/There has been an error./i)).toBeInTheDocument();
	expect(screen.getByText(/Please try again later.../i)).toBeInTheDocument();
	expect(screen.getByAltText(/error/i)).toBeInTheDocument();
});
