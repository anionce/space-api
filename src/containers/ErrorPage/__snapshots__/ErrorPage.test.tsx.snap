// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correcly 1`] = `
<div>
  <div
    class="error-container"
  >
    <h1>
      Ooops!
    </h1>
    <h2>
      There has been an error.
    </h2>
    <h3>
      Please try again later...
    </h3>
    <img
      alt="error"
      src="error.svg"
    />
  </div>
</div>
`;
